### DISCUSSION TITLE
[Question] Using a Transform from code to modify proxy request headers in .net core 3.x application

### DISCUSSION THREAD

**Discussion by IlyasMansoor** [dotnet/yarp#1002] [Q&A] [answered]
I am new to using YARP, It is so very developer friendly and makes to job a lot easier. Thanks to the community!

I am trying to use YARP in a .net core 3.1 application and there is a need to transform the request response to add some custom headers conditionally, When I tried the approach of calling AddTranforms from Startup.ConfigureServices as shown below,

```c#
services.AddReverseProxy()
                .LoadFromConfig(_configuration.GetSection("ReverseProxy"))
                .AddTransforms(
                    builderContext =>
                    {
                        builderContext.AddRequestTransform(
                            async transformContext =>
                            {
                                transformContext.ProxyRequest.Headers.Add("CustomHeader", "CustomValue");
                            });
                    });
```

I get a compilation error as below,
"Property, indexer, or event 'RequestTransformContext.ProxyRequest' is not supported by the language"

I feel this is because the current c# language version is 8.0 as I am using .net core 3.1 framework and the ProxyRequest property is having and init accessor defined which is supported only from langversion 9.0. Is there a way to work around this problem without having to upgrade my framework version to .Net 5? As upgrading the framework version is currently not an option.

**Comment by Tratcher** [comment, marked as answer]
See https://github.com/microsoft/reverse-proxy/issues/837

**Reply by IlyasMansoor** [reply]
Thank you for the redirection, I will cast my vote as well on that page for being blocked as our requirement is also similar and we are at a stage of not being able to upgrade the framework to .Net 5

**Reply by Tratcher** [reply]
We don't require upgrading the framework to 5.0, only the Sdk. The two are independent.
